models/pathology_nuclei_segmentation_classification/scripts/prepare_patches.py

```diff
@@ -0,0 +1,219 @@
+        if h_flag and w_flag:
+            ptx = (im_h - self.patch_size[0], im_w - self.patch_size[1])
+            win = self.__get_patch(x, ptx)
+            sub_patches.append(win)
+        return sub_patches
+
+    def __extract_mirror(self, x):
+        """Extracted patches with mirror padding the boundary such that the
+        central region of each patch is always within the orginal (non-padded)
+        image while all patches' central region cover the whole orginal image.
+
+        Args:
+            x         : input image, should be of shape HWC
+            patch_size  : a tuple of (h, w)
+            step_size : a tuple of (h, w)
+        Return:
+            a list of sub patches, each patch is same dtype as x
+
+        """
+        diff_h = self.patch_size[0] - self.step_size[0]
+        padt = diff_h // 2
+        padb = diff_h - padt
+
+        diff_w = self.patch_size[1] - self.step_size[1]
+        padl = diff_w // 2
+        padr = diff_w - padl
+
+        pad_type = "reflect"
+        x = np.lib.pad(x, ((padt, padb), (padl, padr), (0, 0)), pad_type)
+        sub_patches = self.__extract_valid(x)
+        return sub_patches
+
+    def extract(self, x, patch_type):
+        patch_type = patch_type.lower()
+        self.patch_type = patch_type
+        if patch_type == "valid":
+            return self.__extract_valid(x)
+        elif patch_type == "mirror":
+            return self.__extract_mirror(x)
+        else:
+            raise ValueError(f"Unknown Patch Type {patch_type}")
+
+
+def main(cfg):
+    xtractor = PatchExtractor(cfg["patch_size"], cfg["step_size"])
+
+    for phase in ["Train", "Test"]:
+        img_dir = os.path.join(cfg["root"], f"{phase}/Images")
+        ann_dir = os.path.join(cfg["root"], f"{phase}/Labels")
+
+        file_list = glob.glob(os.path.join(ann_dir, "*mat"))
+        file_list.sort()  # ensure same ordering across platform
+
+        out_dir = f"{cfg['root']}/Prepared/{phase}"
+        if os.path.isdir(out_dir):
+            shutil.rmtree(out_dir)
+        os.makedirs(out_dir)
+
+        pbar_format = "Process File: |{bar}| {n_fmt}/{total_fmt}[{elapsed}<{remaining},{rate_fmt}]"
+        pbarx = tqdm.tqdm(total=len(file_list), bar_format=pbar_format, ascii=True, position=0)
+
+        for file_path in file_list:
+            base_name = pathlib.Path(file_path).stem
+
+            img = load_img(f"{img_dir}/{base_name}.png")
+            ann = load_ann(f"{ann_dir}/{base_name}.mat")
+
+            # *
+            img = np.concatenate([img, ann], axis=-1)
+            sub_patches = xtractor.extract(img, cfg["extract_type"])
+
+            pbar_format = "Extracting  : |{bar}| {n_fmt}/{total_fmt}[{elapsed}<{remaining},{rate_fmt}]"
+            pbar = tqdm.tqdm(total=len(sub_patches), leave=False, bar_format=pbar_format, ascii=True, position=1)
+
+            for idx, patch in enumerate(sub_patches):
+                image_patch = patch[..., :3].transpose(2, 0, 1)  # make channel first
```

Comment:
some of these should go as part of training pre transforms.. like EnsureChannelFirst, Adding Channel for label...
This should only produce the required image/label/type_map files which can be either numpy or png and hence loadable in LoadImage transform.

This will enable others to prepare the similar dataset for additional training.. u have to separate and simplify things wrt ConSeP data and non ConSeP dataset..

Reply:
Hi @SachidanandAlle, thanks for pointing it out, have removed it in this [commit](https://github.com/Project-MONAI/model-zoo/pull/245/commits/ff55e7e8333e1818b30e04ccc4a874f6d86fb95b).